---
title: "RNAseqII.at"
format: html
editor: visual
---

## RNA-seq Part 2

In the first part, we learned how to analyze RNA-seq data and determine which genes in the sample dataset were differentially expressed or turned on/off from instar 1 to instar 2. We used edgeR alongside a couple other R packages to further analyze the sample data set.

RNA-seq part 2 is focusing on visualizing data and narrowing down on genes that are different as well as analysis of the patterns in the data. This part is focusing on a sample data set from the R bioinformatics cookbook using the model plant species Arabidopsis thaliana. This is focusing on how its gene expression varies in different parts of the plant. Then we can use RNA-seq to see if gene expression varies by ecotype.

# Install/Load packages

```{r}
#| warning: false
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

#BiocManager::install("ComplexHeatmap")
```

These packages will allow for data to be organized and visualized with useful color scheme. Installed and then replaced the \#

```{r}
#install.packages("viridisLite")
#install.packages("stringr")
#install.packages("RColorBrewer")
#install.packages("circlize")
```

Loading all the packages including rbioinfocookbook

```{r}
library(ComplexHeatmap)
library(viridisLite)
library(stringr)
library(RColorBrewer)
library(circlize)

library(rbioinfcookbook)
```

# Loading the dataset

What is happening: loading dataset from rbioinfocookbook, name of the dataset is at_tf_gex, pulling a few columns of interest and scale the data to make it more visually useful. Then using a function in the stringer package to split the data by ecotype.

rbioinfocookbook, is spelled wrong and I think is the only reason my R is not throwing an error as if spelled correctly it is masked in global environment by count_dataframe

```{r}
mat <- log(as.matrix(at_tf_gex[ , 5:55]))
ecotype <- stringr::str_split(colnames(mat), ",", simplify = TRUE)[,1]
part <- stringr::str_split(colnames(mat), ",", simplify = TRUE)[,2]
```

# Color Palettes

use circlize and viridisLite packages to create color palettes for different types of data in the heatmap, makes it so we can create a unique color palette for each ecotype and plant part.

```{r}
data_col_func <- circlize::colorRamp2(seq(0, max(mat), length.out = 6), viridisLite::magma(6))

ecotype_colors <- c(RColorBrewer::brewer.pal(12, "Set3"), RColorBrewer::brewer.pal(5, "Set1"))
names(ecotype_colors) <- unique(ecotype)

part_colors <- RColorBrewer::brewer.pal(3, "Accent")
names(part_colors) <- unique(part)
```

\*doing command+shift only runs one line of the code and does not run the whole chunk?

# Annotations and Formatting

Adding annotations to heatmap, ecotype and plant part information are included in the top_annot. use annotation_name_side to set the annotation to the left of the color that they represent.

-   side_annot used to add annot to the rows of the heatmap. adding the length information for the samples in this scenario.

    -   formatting the plot

        -   anno_points, specifies locus of the points to be plotted

        -   pch, shape of the points

        -   size, size of pts

        -   axis_param, locus of the ticks on the x-axis

```{r}
top_annot <- HeatmapAnnotation("Ecotype" = ecotype, "Plant Part" = part, col = list("Ecotype" = ecotype_colors, "Plant Part" = part_colors), annotation_name_side = "left")

side_annot <- rowAnnotation(length = anno_points(at_tf_gex$Length, pch = 16, size = unit(1, "mm"), axis_param = list(at = seq(1, max(at_tf_gex$Length), length.out = 4)),))
```

# Creating the Heatmap

Using the heatmap function in complex heatmap, going to specify mat as the dataset. data_col_func uses the color palettes from the chunk above and the top_annot and side_annot use the annotations made already.

-   Other functions adjust the display of the plot (what the code is going to be doing in next chunk)

    -   row-km: set the \# of clusters for row

    -   cluster_columns, set to TRUE, means column of the heatmap cluster based on similarity in gene expression

    -   column_split: set to ecotype, and groups the columns by ecotype

    -   show_column_names, FALSE, hides the column names to make neater display

    -   column_title: left blank to make the plot look less cluttered

        -   log(TPM) looking at the gene expression by transcript count

        -   lighter colors represent genes more highly expressed.

```{r}
ht_1 <- Heatmap(mat, name="log(TPM)", row_km = 6, col = data_col_func, top_annotation = top_annot, right_annotation = side_annot, cluster_columns = TRUE, column_split = ecotype, show_column_names = FALSE, column_title = " ")

ComplexHeatmap::draw(ht_1)
```

# Lingering Questions/Reflection

1.  Able to notice any trends in gene expression based on ecotype or plant part?

    1.  Ecotype Edi-0 has the plant part colors switched from the peach-purple-green to purple-green-peach and threw me off for a second when looking for a difference. Is it on purpose or is it a mistake?

    2.  The ecotypes all have similar log(TPM) by transcript counts which is interesting in an ecotype that is grown in a specific environment, as long as I am reading this correctly.

2.  Running the individual code chunks did not work for me as it was throwing errors like, did not know what heatmap object was. I definitely need to continue playing with R in basics of how code syntax works as it is interesting.
